frontend/src/app/shared/map/map.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit } from '@angular/core';
+import Map from 'ol/Map';
+import View from 'ol/View';
+import TileLayer from 'ol/layer/Tile';
+import XYZ from 'ol/source/XYZ';
+import { fromLonLat } from 'ol/proj';
+import Feature from 'ol/Feature';
+import Point from 'ol/geom/Point';
+import { Vector as VectorSource } from 'ol/source';
+import { Vector as VectorLayer } from 'ol/layer';
+import Overlay from 'ol/Overlay';
+import { click } from 'ol/events/condition';
+import { Style, Icon } from 'ol/style';
+
+@Component({
+  selector: 'app-map',
+  templateUrl: './map.component.html',
+  styleUrls: ['./map.component.scss']
+})
+export class MapComponent implements OnInit {
+  map!: Map;
+  vectorSource!: VectorSource;
+  vectorLayer!: VectorLayer<any>;
+  overlay!: Overlay;
+
```

Comment:
Here an `EventEmitter` is missing which emits the value of the clicked coordinates